### PR TITLE
Migrate away from JSS

### LINKED ISSUE #636: Upgrade Material UI to 5
This is the latest stable version per https://mui.com/getting-started/usage/

Per [migration guide](https://mui.com/guides/migration-v4), we need to:

- [x] Upgrade React to v17
- [x] Upgrade (if needed) react-scripts, @types/react, @types/react-dom
- [x] Setup ThemeProvider properly
- [x] Update MUI version
- [x] Run codemods
- [x] Handle individual breaking changes
  - [x] Theme structure
  - [x] [Pickers](https://mui.com/guides/pickers-migration/)
- [x] #847

### PR DESCRIPTION
**Pull request by josueetcom**
## Motivation and Context

Fixes #636. Last of the migration PRs. Changed some things slightly but should still work.

## How Has This Been Tested?

Existing tests pass. Also manually checked to make sure things still looked ok.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] User visible change (users will notice UI or functional changes)

## Previewing Changes

1. Go to render preview URL below.
2. Click around and make sure nothing's broken.

## Checklist:

- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by josueetcom** [comment]
I suggest reviewing ignoring whitespace

File: src/components/AmountFilter.jsx [hunk 1]
```diff
@@ -34,7 +18,7 @@ export default function AmountFilter(props) {
     <>
       <Button
         variant="outlined"
-        className={classes.buttonStyle}
+        sx={{ margin: 1, height: (theme) => theme.spacing(4) }}
```
Comment (edwingl10) [inline src/components/AmountFilter.jsx:21]: margin can be written as `m`

File: src/components/AmountFilter.jsx [hunk 2]
```diff
@@ -47,19 +31,25 @@ export default function AmountFilter(props) {
         anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
         transformOrigin={{ vertical: 'top', horizontal: 'left' }}>
         <Grid container>
-          <Grid item className={classes.filterStyle}>
-            <InputLabel className={classes.labelStyle}>Min Amount</InputLabel>
+          <Grid item sx={{ margin: 1 }}>
+            <InputLabel sx={{ paddingLeft: 1 }}>Min Amount</InputLabel>
```
Comment (edwingl10) [inline src/components/AmountFilter.jsx:35]: you can simply write`pl`

File: src/components/AmountFilter.jsx [hunk 3]
```diff
@@ -47,19 +31,25 @@ export default function AmountFilter(props) {
...
               error={error}
               value={min || ''}
               onChange={onMinChange}
             />
           </Grid>
-          <Grid item className={classes.dashStyle}>
+          <Grid
+            item
+            sx={{
+              paddingTop: 2,
```
Comment (edwingl10) [inline src/components/AmountFilter.jsx:45]: `pt`

Comment (edwingl10) [inline src/components/AmountFilter.jsx:45]: `pt:2`

File: src/components/Header.jsx [hunk 4]
```diff
@@ -123,15 +107,22 @@ const languages = {
...
   return (
     <Container>
       <UnderConstructionAlert t={t} />
       <OnRenderSnackbar />
-      <Grid container className={classes.header} spacing={3}>
+      <Grid
+        container
+        sx={{
+          flexGrow: 1,
+          padding: 3,
```
Comment (edwingl10) [inline src/components/Header.jsx:121]: `p:3`

**Review by edwingl10 — DISMISSED** [review]

File: src/components/Header.jsx (hunk 4 above)
Comment (josueetcom) [reply src/components/Header.jsx:121]: I'm going to keep this one as there's lots of styling going on here and the shorthand doesn't help condense this into one line :P

**Review by edwingl10 — APPROVED** [review]

**Pull request merged**